### PR TITLE
SuperBuild - Optimizations + More (FeroxRev) - WIP

### PR DESCRIPTION
**Pull request by CDAGaming**
## Features
- All Dependencies Updated to Latest (No Issues @jjskuld, according to several tests)
- Many Typos have been Fixed
- Many Optimizations/Simplified Code-Behind for a Faster Experience
- Support for iOS Devices Up to 10.3
- Support for API 0.59.1 & 0.59.2 (Already in Official)
- License is now Properly an Embedded Resource so it is not visible
- Adds More Extended PokeEase Integration
- Fixes GUI Settings Descriptions (Empty or with typos, WIP)

## Known Bugs
- Sometimes, Going on the Hub/Browser Tab, the DotNetBrowser would take up the full screen
Workaround: Resize the screen a little bit for it to correct itself
- Sometimes, an error like the one below may appear when using 0.59 API (In Progress):
`[21:20:03] (ERROR) Hashing Servers errors, throttling calls. SESSION INVALIDATED EXCEPTION
[21:20:04] (ERROR) Resetting position before relogging in.`

## Requirements
- If you are Testing/Building this Source, Please don't forget to Get the other Pull Requests for the Submodules (FeroxRev)

## Downloads
Build 0401 - Coming in One Hour!
Build 0330 - [Click Here](https://mega.nz/#!3MhETZrK!m_H0g9QD7X9HgTxfJtLHR049kJfSELaZGTIr95nn88Q)

## Related PRs
[NecroBot](https://github.com/Necrobot-Private/NecroBot/pull/1410)

### CODE REVIEW

**Review by jjskuld — CHANGES_REQUESTED** [review]
So this pull is a revert of pull #92. Take a look at the changes from https://github.com/Necrobot-Private/PokemonGo.RocketAPI/pull/92/files.

Pull #92 needs to be applied on top of these changes before we merge this.

Files that need to be fixed:
Authentication/Session.cs 
- Add lines 5, 8 from left
- Remove line 11 from right
- Revert line 41 from right
- Revert line 50 from right

Extensions/HttpClientExtensions.cs
The change to the out parameter is breaking the build (at least in VS2015).
- Revert line 163 from right

Client.cs 
- Revert change to Line 144

Hash/PokefarmerHasher.cs
- Fix messed up spacing in lines 19-21 (you can see the difference in GitHub)
- Add line 28 back from the left
- Fix messed up spacing in lines 107-114
- Fix messed up spacing in lines 147-153
- Add back missing line 160 from left (which is probably going to cause build break as well).

Ack.. I'm going to stop here.. this is taking too long to diff.
Bottom line, need to apply pull #92.

**Comment by jjskuld** [comment]
Ok, what i'm going to do is to try to cherry pick your changes on top of a clean branch of master. Maybe easier than the mess we have here.

**Comment by jjskuld** [comment]
BTW this is not building even in VS 2017 due to Pull #92 being reverted.

screen shot 2017-04-02 at 12 15 43 am

Trying to fix it now.

**Comment by jjskuld** [comment]
Closing this in favor of Pull #99
